### PR TITLE
feat(RDS): add resource rds lts log

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:
add resource rds lts log
**Special notes for your reviewer**:

**Release note**:

```release-note
add resource rds lts log
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/rds/ TESTARGS='-run TestAccRdsLtsLog_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccRdsLtsLog_basic -timeout 360m -parallel 4
=== RUN   TestAccRdsLtsLog_basic
=== PAUSE TestAccRdsLtsLog_basic
=== CONT  TestAccRdsLtsLog_basic
--- PASS: TestAccRdsLtsLog_basic (617.41s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       617.457s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		return nil, fmt.Errorf("error retrieving RDS LTS config: the LTS config can not be found")
+	}
+
+	return ltsConfig, nil
+}
+
+func TestAccRdsLtsLog_basic(t *testing.T) {
+	var obj interface{}
+	rName := acceptance.RandomAccResourceName()
+	pwd := fmt.Sprintf("%s%s%d", acctest.RandString(5), acctest.RandStringFromCharSet(2, "!#%^*"),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go]: how about use acceptance.RandomPassword

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+	var getRdsLtsLogRespBody interface{}
+	err = json.Unmarshal(getRdsLtsLogRespJson, &getRdsLtsLogRespBody)
+	if err != nil {
+		return diag.Errorf("error unmarshaling RDS LTS configs: %s", err)
+	}
+
+	jsonPath := fmt.Sprintf("instance_lts_configs[?lts_configs[?log_type=='%s']]|[0].lts_configs|[0]", parts[2])
+	ltsConfig := utils.PathSearch(jsonPath, getRdsLtsLogRespBody, nil)
+	if !utils.PathSearch("enabled", ltsConfig, false).(bool) {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go:184]:
```suggestion
		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error retrieving RDS LTS config")
```

File: docs/resources/rds_lts_log.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+* `lts_stream_id` - (Required, String, ForceNew) Specifies the ID of the LTS log stream.
+  Changing this creates a new resource.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attribute is exported:
+
+* `id` - The resource ID in format of `<engine>/<instance_id>/<log_type>`.
+It is composed of the RDS instance engine, the RDS instance ID and the LTS log type.
```
Comment (Jason-Zhang9309) [inline docs/resources/rds_lts_log.md:52]:
```suggestion
  It is composed of the RDS instance engine, the RDS instance ID and the LTS log type.
```

Comment (Jason-Zhang9309) [inline docs/resources/rds_lts_log.md]:
```suggestion
* `id` - The resource ID in format of `<engine>/<instance_id>/<log_type>`.
  It is composed of the RDS instance engine, the RDS instance ID and the LTS log type.
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go (hunk 1 above)
Comment (346010762) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go]: acceptance.RandomPassword is ok.

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go:184]: the error message is useless when status code is 404.

File: docs/resources/rds_lts_log.md (hunk 3 above)
Comment (346010762) [reply docs/resources/rds_lts_log.md]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+)
+
+// @API RDS POST /v3/{project_id}/{engine}/instances/logs/lts-configs
+// @API RDS GET /v3/{project_id}/{engine}/instances/logs/lts-configs
+// @API RDS DELETE /v3/{project_id}/{engine}/instances/logs/lts-configs
+func ResourceRdsLtsLog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRdsLtsLogCreate,
+		ReadContext:   resourceRdsLtsLogRead,
+		DeleteContext: resourceRdsLtsLogDelete,
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]: add updateContext,  lts_group_id and lts_stream_id can be updated

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+	ltsConfig := utils.PathSearch(jsonPath, getRdsLtsLogRespBody, nil)
+	if !utils.PathSearch("enabled", ltsConfig, false).(bool) {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("engine", parts[0]),
+		d.Set("instance_id", parts[1]),
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]:
```suggestion
		d.Set("engine", utils.PathSearch("instance_lts_configs[0].instance.engine_name", nil)
		d.Set("instance_id", utils.PathSearch("instance_lts_configs[0].instance.id", nil)
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+		WaitTarget:   []string{"ACTIVE"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error associating RDS with LTS log: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", engine, instanceID, logType))
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]:
```suggestion
	d.SetId(fmt.Sprintf("%s/%s",  instanceID, logType))
```
the instance engine can be get by the instance id

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+// @API RDS POST /v3/{project_id}/{engine}/instances/logs/lts-configs
+// @API RDS GET /v3/{project_id}/{engine}/instances/logs/lts-configs
+// @API RDS DELETE /v3/{project_id}/{engine}/instances/logs/lts-configs
+func ResourceRdsLtsLog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRdsLtsLogCreate,
+		ReadContext:   resourceRdsLtsLogRead,
+		DeleteContext: resourceRdsLtsLogDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]: should use custom import func and get the instance engine by instance id

File: huaweicloud/services/rds/common.go [hunk 8]
```diff
@@ -71,6 +72,23 @@ func handleMultiOperationsError(err error) (bool, error) {
 			return true, err
 		}
 	}
+	if errCode, ok := err.(golangsdk.ErrDefault500); ok {
+		var apiError interface{}
+		if jsonErr := json.Unmarshal(errCode.Body, &apiError); jsonErr != nil {
+			return false, fmt.Errorf("unmarshal the response body failed: %s", jsonErr)
+		}
+
+		errorCode, errorCodeErr := jmespath.Search("errCode||error_code", apiError)
```
Comment (houpeng80) [inline huaweicloud/services/rds/common.go]:
```suggestion
		errorCode, errorCodeErr := jmespath.Search("error_code", apiError)
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_lts_group" "test" {
+  group_name  = "%[2]s"
+  ttl_in_days = 1
+}
+
+resource "huaweicloud_lts_stream" "test" {
+  depends_on  = [huaweicloud_lts_group.test]
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go]: it is useless

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go (hunk 4 above)
Comment (346010762) [reply huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go (hunk 5 above)
Comment (346010762) [reply huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go (hunk 6 above)
Comment (346010762) [reply huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go (hunk 7 above)
Comment (346010762) [reply huaweicloud/services/rds/resource_huaweicloud_rds_lts_log.go]: done

File: huaweicloud/services/rds/common.go (hunk 8 above)
Comment (346010762) [reply huaweicloud/services/rds/common.go]: done

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go (hunk 9 above)
Comment (346010762) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_lts_log_test.go]: done

**Pull request merged**